modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OIDCUserInfoProcessor.java

```diff
@@ -482,6 +504,46 @@ private String _getClaimString(
 		return (String)claimObject;
 	}
 
+	private String _getFallbackMatcher(
+			String authServerWellKnownURI, String clientId, long companyId,
+			String issuer, String tokenEndpoint)
+		throws Exception {
+
+		String generatedLocalWellKnownURI =
+			OAuthClientEntryUtil.generateLocalWellKnownURI(
+				issuer, tokenEndpoint);
+
+		String filterString = null;
+
+		if (authServerWellKnownURI.equals(generatedLocalWellKnownURI)) {
+			filterString = StringBundler.concat(
+				"(&(companyId=", companyId, ")(tokenEndpoint=", tokenEndpoint,
+				")(issuerURL=", issuer, ")(openIdConnectClientId=", clientId,
+				"))");
+		}
+		else {
+			filterString = StringBundler.concat(
+				"(&(companyId=", companyId, ")(discoveryEndpoint=",
+				authServerWellKnownURI, ")(openIdConnectClientId=", clientId,
+				"))");
+		}
+
+		Configuration[] configurations = _configurationAdmin.listConfigurations(
+			filterString);
+
+		String fallbackMatcher = "email";
```

Comment:
Instead of:

```
        String fallbackMatcher = "email";

		if (configurations != null) {
			Dictionary<String, Object> properties =
				configurations[0].getProperties();

			fallbackMatcher = GetterUtil.getString(
				properties.get("fallbackMatcher"));
		}

		return fallbackMatcher;
 ```

do this:

```
		if (configurations == null) {
			return "email";
		}

         Dictionary<String, Object> properties =
				configurations[0].getProperties();

		return GetterUtil.getString(
				properties.get("fallbackMatcher"));
 ```